Add inter-service auth sample

ace-n opened the pull request:
```release-note:none
```

Adds a sample for [Service-to-Service auth](https://cloud.google.com/run/docs/authenticating/service-to-service).

@pattishin FYI - I've used a default container (`us-docker.pkg.dev/cloudrun/container/hello`) and told end-users (via comments) to replace these with _their own_ containers.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6248-old..auto-pr-6248) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-6248-old..auto-pr-6248) ( 4 files changed, 183 insertions(+))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 88 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2095` 
Passed tests `1865` 
Skipped tests: `226` 
Failed tests: `4` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccCloudRunService_cloudRunServiceInterserviceExample|TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample|TestAccSqlUser_mysqlDisabled </details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccSqlUser_mysqlDisabled`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/recording/TestAccSqlUser_mysqlDisabled.log)] 

Tests failed during RECORDING mode:
 `TestAccCloudRunService_cloudRunServiceInterserviceExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/recording/TestAccCloudRunService_cloudRunServiceInterserviceExample.log)]
`TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/recording/TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation.log)]
`TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/recording/TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/579713e7-1b36-4fbc-b933-c447c4eaf64c/recording) for each test

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6248-old..auto-pr-6248) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-6248-old..auto-pr-6248) ( 4 files changed, 183 insertions(+))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 88 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2095` 
Passed tests `1865` 
Skipped tests: `226` 
Failed tests: `4` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample|TestAccFirebaserulesRelease_BasicRelease|TestAccCloudRunService_cloudRunServiceInterserviceExample </details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccCloudRunService_cloudRunServiceInterserviceExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/recording/TestAccCloudRunService_cloudRunServiceInterserviceExample.log)]
`TestAccFirebaserulesRelease_BasicRelease`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/recording/TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation.log)]
`TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/recording/TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/0d67a818-d328-4e5f-ad4b-35670a43fb9b/recording) for each test

ace-n commented:
I didn't see an auto-assigned reviewer (perhaps because I added @pattishin?), so I added @c2thorn as well.

**Test failures are unrelated**, so this _should_ be good to merge. 🤞

pattishin commented on mmv1/templates/terraform/examples/cloud_run_service_interservice.tf.erb:
```diff
@@ -0,0 +1,88 @@
+# Example of using a public Cloud Run service to call a private one
+
+# [START cloudrun_service_interservice_public]
```
Will need separate region tags for each resource

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 112 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6248-old..auto-pr-6248) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-6248-old..auto-pr-6248) ( 4 files changed, 183 insertions(+))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-6248-old..auto-pr-6248) ( 1 file changed, 92 insertions(+))

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/2d3f0b8e-154f-4c9f-aee1-17814038d40e/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/2d3f0b8e-154f-4c9f-aee1-17814038d40e/recording/TestAccPrivatecaCertificateAuthority_subordinateCaActivatedByFirstPartyIssuerOnCreation.log)]
`TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/2d3f0b8e-154f-4c9f-aee1-17814038d40e/recording/TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample.log)] 

All tests passed
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/2d3f0b8e-154f-4c9f-aee1-17814038d40e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6248/artifacts/2d3f0b8e-154f-4c9f-aee1-17814038d40e/recording) for each test
